### PR TITLE
Apply jacoco plugin

### PR DESCRIPTION
**Pull request by alovew**
## What
apply jacoco analysis to all classes and generate a report. we want to display the code coverage errors in the build, but not to fail the build.

### CODE REVIEW

**Comment by benmoriceau** [comment]
Do you know what will be the impact on the build time? Does it seems to have a big impact if you run gradle with the `--scan` option?

File: build.gradle [hunk 1]
```diff
@@ -215,6 +215,37 @@ subprojects {
...
+            csv.required = false
+        }
+    }
+
+    jacocoTestCoverageVerification {
+        violationRules {
+            failOnViolation = false
+            rule {
+                element = 'CLASS'
+                excludes = ['**/*Test*', '**/generated*']
```
Comment (jdpgrailsdev) [inline build.gradle:236]: We may need to revisit this.  In cloud, we had to change the location of the generated sources to consistently use the same location to make this effective.  For now, this is fine as an experiment to see how much of the generated source gets excluded.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Comment by jdpgrailsdev** [comment]
> Do you know what will be the impact on the build time? Does it seems to have a big impact if you run gradle with the `--scan` option?

@benmoriceau It should have minimal impact.  It will generate a coverage file per project, which should be on the order of milliseconds to seconds for each.  As you suggest, running with `--scan` will confirm.

**Comment by alovew** [comment]
@benmoriceau this is what I'm seeing (on an m1):
BUILD SUCCESSFUL in 3m 15s
342 actionable tasks: 77 executed, 265 up-to-date

very similar to what I see on master

**Pull request merged**